<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Serial</class>
 <widget class="QWidget" name="Serial">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>929</width>
    <height>810</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="11" column="1" colspan="2">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>geneave 0.2 - arduino weaver control - development version Ignacio Gutierrez &lt;ignacioandcorral@yahoo.com&gt;, Wolfram Teetz &lt;wolframteetz@gmail.com&gt; </string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="6" column="1" rowspan="2" colspan="2">
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="10" column="1" colspan="2">
    <widget class="Line" name="line_3">
     <property name="sizePolicy">
      <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="4" column="1" colspan="2">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Servo setup</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0" colspan="3">
       <widget class="QTableWidget" name="tableWidgetAmounts">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>30</number>
        </attribute>
        <row>
         <property name="text">
          <string>0</string>
         </property>
        </row>
        <row>
         <property name="text">
          <string>1</string>
         </property>
        </row>
        <column>
         <property name="text">
          <string>1</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>2</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>3</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>4</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>5</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>6</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>7</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>8</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>9</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>10</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>11</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>12</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>13</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>14</string>
         </property>
        </column>
        <item row="0" column="0">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="1">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="2">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="3">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="4">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="5">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="6">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="7">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="8">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="9">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="10">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="11">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="12">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="0" column="13">
         <property name="text">
          <string>00</string>
         </property>
        </item>
        <item row="1" column="0">
         <property name="text">
          <string>18</string>
         </property>
        </item>
        <item row="1" column="1">
         <property name="text">
          <string>18</string>
         </property>
        </item>
        <item row="1" column="2">
         <property name="text">
          <string>18</string>
         </property>
        </item>
        <item row="1" column="3">
         <property name="text">
          <string>18</string>
         </property>
        </item>
        <item row="1" column="4">
         <property name="text">
          <string>18</string>
         </property>
        </item>
        <item row="1" column="5">
         <property name="text">
          <string>18</string>
         </property>
        </item>
        <item row="1" column="6">
         <property name="text">
          <string>18</string>
         </property>
        </item>
        <item row="1" column="7">
         <property name="text">
          <string>16</string>
         </property>
        </item>
        <item row="1" column="8">
         <property name="text">
          <string>23</string>
         </property>
        </item>
        <item row="1" column="9">
         <property name="text">
          <string>16</string>
         </property>
        </item>
        <item row="1" column="10">
         <property name="text">
          <string>16</string>
         </property>
        </item>
        <item row="1" column="11">
         <property name="text">
          <string>16</string>
         </property>
        </item>
        <item row="1" column="12">
         <property name="text">
          <string>18</string>
         </property>
        </item>
        <item row="1" column="13">
         <property name="text">
          <string>18</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QToolButton" name="toolButtonAddRowServo">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>add row</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QToolButton" name="toolButtonRemoveRowServo">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>remove row</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="5" column="1" colspan="2">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Pattern</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="0" colspan="4">
       <widget class="QTableWidget" name="tableWidgetPattern">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>30</number>
        </attribute>
        <row>
         <property name="text">
          <string>1</string>
         </property>
        </row>
        <row>
         <property name="text">
          <string>2</string>
         </property>
        </row>
        <row>
         <property name="text">
          <string>3</string>
         </property>
        </row>
        <row>
         <property name="text">
          <string>4</string>
         </property>
        </row>
        <row>
         <property name="text">
          <string>5</string>
         </property>
        </row>
        <column>
         <property name="text">
          <string>1</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>2</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>3</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>4</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>5</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>6</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>7</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>8</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>9</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>10</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>11</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>12</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>13</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>14</string>
         </property>
        </column>
        <item row="0" column="0">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="0" column="1">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="0" column="2">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="0" column="3">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="0" column="4">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="0" column="5">
         <property name="text">
          <string>1</string>
         </property>
        </item>
        <item row="0" column="6">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="0" column="7">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="0" column="8">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="0" column="9">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="0" column="10">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="0" column="11">
         <property name="text">
          <string>1</string>
         </property>
        </item>
        <item row="0" column="12">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="0" column="13">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="1" column="0">
         <property name="text">
          <string>1</string>
         </property>
        </item>
        <item row="1" column="1">
         <property name="text">
          <string>1</string>
         </property>
        </item>
        <item row="1" column="2">
         <property name="text">
          <string>1</string>
         </property>
        </item>
        <item row="1" column="3">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="1" column="4">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="1" column="5">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="1" column="6">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="1" column="7">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="1" column="8">
         <property name="text">
          <string>1</string>
         </property>
        </item>
        <item row="1" column="9">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="1" column="10">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="1" column="11">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="1" column="12">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="1" column="13">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="0">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="1">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="2">
         <property name="text">
          <string>1</string>
         </property>
        </item>
        <item row="2" column="3">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="4">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="5">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="6">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="7">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="8">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="9">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="10">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="11">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="12">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="2" column="13">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="0">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="1">
         <property name="text">
          <string>1</string>
         </property>
        </item>
        <item row="3" column="2">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="3">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="4">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="5">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="6">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="7">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="8">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="9">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="10">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="11">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="12">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="3" column="13">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="0">
         <property name="text">
          <string>1</string>
         </property>
        </item>
        <item row="4" column="1">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="2">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="3">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="4">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="5">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="6">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="7">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="8">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="9">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="10">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="11">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="12">
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item row="4" column="13">
         <property name="text">
          <string>0</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QToolButton" name="toolButtonAddRowPattern">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>add row</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QToolButton" name="toolButtonRemoveRowPattern">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>remove row(s)</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="1" colspan="2">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Serial Port</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="1">
       <widget class="QComboBox" name="comboBoxSerialPort">
        <property name="editable">
         <bool>true</bool>
        </property>
        <item>
         <property name="text">
          <string>COM3</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>/dev/tty.usbmodemFA141</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Serial port</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="8" column="1" colspan="2">
    <widget class="QGroupBox" name="groupBox_5">
     <property name="title">
      <string>Servo Control</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_6">
      <item row="0" column="3">
       <widget class="QPushButton" name="pushButtonExecute">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Weave all rows</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="pushButtonWeaveSelectedRows">
        <property name="text">
         <string>Weave selected rows</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_Times">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Time(s)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QSpinBox" name="spinBoxTimesExecute">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>99999</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="4">
       <widget class="QTextBrowser" name="textBrowserStatus">
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'.SF NS Text'; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Ready to weave.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1" colspan="2">
    <widget class="QGroupBox" name="groupBox_4">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Control</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="0" column="3">
       <widget class="QPushButton" name="pushButtonSave">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Save</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="pushButtonLoad">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Load</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="labelName">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>CustomWeave</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="pushButtonSaveAs">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Save As...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
